tm2/pkg/sdk/baseapp.go

```diff
@@ -652,11 +652,9 @@ func (app *BaseApp) runMsgs(ctx Context, msgs []Msg, mode RunTxMode) (result Res
 		// Each message result's Data must be length prefixed in order to separate
 		// each result.
 		data = append(data, msgResult.Data...)
-		events = append(events, msgResult.Events...)
```

Comment:
you shouldn't remove this from the slice `msgResult.Events...`

Reply:
102c0d663cbf4d37c5a0ba7e1bd6966e8cad7b2a